### PR TITLE
remove okteto cloud as context option

### PR DESCRIPTION
**Pull request by AdrianPedriza**
# Proposed changes

- remove okteto cloud as context option

## How to validate

Run `okteto context` and check if okteto cloud context is displayed. It shouldn't

Context shown shouldn't display cloud context:
Captura de pantalla 2023-11-29 a las 13 23 00

### CODE REVIEW

**Comment by AdrianPedriza** [comment]
> @AdrianPedriza could you add a screenshot?

👍 I've updated PR description with it

**Comment by pchico83** [comment]
@rlamana unrelated to this PR, but should we add a link to the SH trial when running `okteto ctx`?

**Comment by andreafalzetti** [comment]
When I run `okteto ctx` on a fresh machine I get an empty line.

Screenshot 2023-11-29 at 16 06 50

Quickest way to reproduce is

`OKTETO_HOME=~/dev okteto ctx` (nb. `~/dev` must exist but not contian the `.okteto` and `.kube` folders)

**Comment by AdrianPedriza** [comment]
> When I run `okteto ctx` on a fresh machine I get an empty line.
> 
> Screenshot 2023-11-29 at 16 06 50
> 
> Quickest way to reproduce is
> 
> `OKTETO_HOME=~/dev okteto ctx` (nb. `~/dev` must exist but not contian the `.okteto` and `.kube` folders)

a6bb86f124657ab33548fea5c439b2e3c7e14d7e

Captura de pantalla 2023-11-29 a las 17 41 51

**Comment by AdrianPedriza** [comment]
> > @rlamana unrelated to this PR, but should we add a link to the SH trial when running `okteto ctx`?
> 
> I think `okteto context` is a potential friction point to onboard new developers so I agree with adding link to docs (explaining what context is, for example, and what value to use)
> 
> Also, when I have no contexts, the copy `Select the context you want to use` could be reviewed 🤔 .
> 
> Another comment about usability: When I have no contexts, why not showing directly the "Enter your Okteto URL" prompt that appears after pressing Enter?
> 
> I think there is an extra step of choosing "Create new context" which is the one and single option for me as a user.

good points. @rlamana what should be the behavior when there is no context added. Ask directly for `Enter your Okteto URL` instead show a list with only `Create new context` entry?

**Comment by AdrianPedriza** [comment]
> > > @rlamana unrelated to this PR, but should we add a link to the SH trial when running `okteto ctx`?
> > 
> > 
> > I think `okteto context` is a potential friction point to onboard new developers so I agree with adding link to docs (explaining what context is, for example, and what value to use)
> > Also, when I have no contexts, the copy `Select the context you want to use` could be reviewed 🤔 .
> > Another comment about usability: When I have no contexts, why not showing directly the "Enter your Okteto URL" prompt that appears after pressing Enter?
> > I think there is an extra step of choosing "Create new context" which is the one and single option for me as a user.
> 
> good points. @rlamana what should be the behavior when there is no context added. Ask directly for `Enter your Okteto URL` instead show a list with only `Create new context` entry? that way we will remove the selector in that case

**Comment by codyjlandstrom** [comment]
> I also don't like the message "A context defines the default cluster/namespace for any Okteto CLI command." We mention "namespace", shouldn't we say something like "A context defines the default Okteto instance or cluster for any Okteto CLI command."?

Agree with this messaging. I think your suggestion makes the most sense.

> And the "Create new context", should it be "Add new context" instead?

I think "Add new context" is the right term as well since you are adding the grouping of credentials, vs creating infrastructure.

**Comment by AdrianPedriza** [comment]
According to what has been discussed, the behavior is as follows:
- **CASE: There is no added context**
Captura de pantalla 2023-12-01 a las 17 48 59

I have slightly modified the original message which was `Enter the context of your Okteto instance or cluster. To install your own instance follow these steps: https://www.okteto.com/free-trial/` to `Enter the URL of your Okteto instance. To install your own instance follow these steps: https://www.okteto.com/free-trial/:` since asking for the URL is the message we have today to ask for the URL of the instance when user select `Add new context`. I think it is more concise and consistent. 

- **CASE: There is already added context**
Captura de pantalla 2023-12-01 a las 17 49 22

I have modified the text of the selector by [the proposed one](https://github.com/okteto/okteto/pull/4073#:~:text=or%20something%20similar.-,I%20also%20don%27t%20like%20the%20message%20%22A%20context%20defines%20the%20default%20cluster/namespace%20for%20any%20Okteto%20CLI%20command.%22%20We%20mention%20%22namespace%22%2C%20shouldn%27t%20we%20say%20something%20like%20%22A%20context%20defines%20the%20default%20Okteto%20instance%20or%20cluster%20for%20any%20Okteto%20CLI%20command.%22%3F,-And%20the%20%22Create): `A context defines the default cluster/namespace for any Okteto CLI command` replaced by `A context defines the default Okteto instance or cluster for any Okteto CLI command.`

- **CASE: There are already contexts, select the option "Add new context".**
Captura de pantalla 2023-12-01 a las 17 49 37

I have changed the text `Context 'Create new context' selected` to `Option 'Add new context' selected`. I think it is more appropriate to speak of an option in the selector.

WDYT? @rlamana @codyjlandstrom

**Comment by codyjlandstrom** [comment]
Sounds great to me @AdrianPedriza

**Comment by rlamana** [comment]
Great, I would however refine a bit more the messages:

- CASE: There is no added context
I would still mention "or cluster" as the CLI still works with vanilla Kubernetes clusters. What about something like this:
```
 ?   Enter the URL of your Okteto instance or a Kuberntes context.
 i   Don't have an Okteto instance? Start by installing Okteto on your Kubernetes cluster: https://www.okteto.com/free-trial/
```

- CASE: There is already added context
I would again add the Kubernetes cluster mention here and change a bit the wording:
```
A context defines the default Okteto instance or Kubernetes context to use for any Okteto CLI command.
```

- CASE: There are already contexts, select the option "Add new context".
shouldn't we mention Kubernetes context here too?
```
 ?   Enter the URL of your Okteto instance or a Kuberntes context [e.g. https://staging.okteto.dev]:
 ```
I would add the "e.g." before the example because otherwise it looks like a default value instead of an example.

cc. @AdrianPedriza @pchico83 @codyjlandstrom

**Comment by AdrianPedriza** [comment]
@rlamana

 **CASE: There is no added context** if we show an INFO log after a QUESTION log I think it's weird, wouldn't it be first the info and then the question to let the user enter the input?
 
 **CASE: There is already added context's suggestion** 👍 

**CASE: There are already contexts, select the option "Add new context".** in this case user only have the possibility to add an okteto context using the okteto instance URL. Kubernetes contexts are not added through this command

**Review by jLopezbarb — APPROVED** [review]

**Comment by AdrianPedriza** [comment]
> ```
> A context defines the default Okteto instance or Kubernetes context to use for any Okteto CLI command.
> i   Don't have an Okteto instance? Start by installing Okteto on your Kubernetes cluster: https://www.okteto.com/free-trial/
> ?   Enter the URL of your Okteto instance:
> ```
message `A context defines the default Okteto instance or Kubernetes context to use for any Okteto CLI command.` is part of the selector header so when we ask for an URL, it disappears from the terminal

with latest changes and suggestions the command out is:
- when there is no okteto context added:
Captura de pantalla 2023-12-05 a las 16 58 07
- when there is an okteto context added and select to add a new one:
Captura de pantalla 2023-12-05 a las 17 02 15

**Review by andreafalzetti — APPROVED** [review]

**Comment by rlamana** [comment]
@codyjlandstrom @pchico83 can you review the last changes to the messages? in this comment: 
https://github.com/okteto/okteto/pull/4073#issuecomment-1841103565

File: cmd/context/use.go [hunk 1]
```diff
@@ -31,7 +31,9 @@ import (
 )
 
 const (
-	personalAccessTokenURL = "https://www.okteto.com/docs/cloud/personal-access-tokens/"
+	personalAccessTokenURL          = "https://www.okteto.com/docs/cloud/personal-access-tokens/"
+	suggestInstallOktetoSH          = "Don't have an Okteto instance? Start by installing Okteto on your Kubernetes cluster: https://www.okteto.com/free-trial/"
```
[the remark this answers is not in the document]
Comment (rlamana) [reply cmd/context/use.go]: LGTM 👍

**Review by rlamana — APPROVED** [review]

**Pull request merged**